### PR TITLE
snark-artifacts starter repo with initial NPM package

### PR DESCRIPTION
**Pull request by artwyman**
Includes the repo skeleton, documentation, and first published version of the proto-pod-gpc artifacts.  Initial working version is 0.0.2 since I had to fix a bug with the packaging of 0.0.1.

Note this PR is to merge into the pre-release branch, not main.

Use of this repo is tied to a PR in zupass: https://github.com/proofcarryingdata/zupass/pull/1708

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,78 @@
...
 
-TODO(POD-P3): Document how to find and use artifacts.
+For testing, you can clone this repo directly to obtain artifacts.  Artifacts
+can also downloaded directly in a browser from this repo using a URL like the
+example below, where "rev" can be a git branch, tag, or commit hash.
+
+  https://raw.githubusercontent.com/proofcarryingdata/snark-artifacts/rev/packages/proto-pod-gpc/proto-pod-gpc_1o-5e-6md-0x0l-0x0t-vkey.json
+
+Officially released artifacts are available as NPM packages accessible as
+devDependencies.  They can also be downloaded from there using an NPM-based
```
Comment (ichub) [inline README.md]:
```suggestion
`devDependencies`.  They can also be downloaded from npm using an NPM-based
```

File: README.md [hunk 2]
```diff
@@ -1,10 +1,78 @@
...
+CDN such as unpkg.com:
+
+  https://unpkg.com/@pcd/proto-pod-gpc-artifacts@0.0.2/proto-pod-gpc_1o-5e-6md-0x0l-0x0t-vkey.json
+
+The `@pcd/gpc` package in the Zupass needs a root URL to find artifacts by
```
Comment (ichub) [inline README.md]:
```suggestion
The `@pcd/gpc` package in the Zupass repo needs a root URL to find artifacts by
```

File: README.md [hunk 3]
```diff
@@ -1,10 +1,78 @@
...
+parent folder and have write access to each.
+
+  cd zupass/packages/lib/gpcircuits
+  yarn gen-test-artifacts
+  yarn copy-test-to-snark-artifacts
+  cd ../../../../snark-artifacts
+  git checkout -b your-experimental-branch-name
+  git commit . -m "test artifacts"
+  git push
+
```
Comment (ichub) [inline README.md]:
````suggestion
```sh
cd zupass/packages/lib/gpcircuits
yarn gen-test-artifacts
yarn copy-test-to-snark-artifacts
cd ../../../../snark-artifacts
git checkout -b your-experimental-branch-name
git commit . -m "test artifacts"
git push
```
````

File: README.md [hunk 4]
```diff
@@ -1,10 +1,78 @@
...
+The `@pcd/gpc` package in the Zupass needs a root URL to find artifacts by
+either of these download mechanism.  To create an appropriate URL you can
+use the `gpcArtifactDownloadURL` function available
+[here](https://github.com/proofcarryingdata/zupass/blob/main/packages/lib/gpc/src/gpc.ts).
+
+## Testing experimental artifacts
+
+The direct GitHub download method above can be used for testing with
+experimental artifacts during circuit development.  These commands assume
+you've cloned the `zupass` and `snark-artifacts` directories in a single
```
Comment (ichub) [inline README.md]: beyond just cloning i think there are probably a few other steps that would be good to mention, e.g. that you need to follow the installation instructions in the repos, etc.

File: README.md [hunk 5]
```diff
@@ -1,10 +1,78 @@
...
+  cd zupass/packages/lib/gpcircuits
+  yarn gen-test-artifacts
+  yarn copy-test-to-snark-artifacts
+  cd ../../../../snark-artifacts
+  git checkout -b your-experimental-branch-name
+  git commit . -m "test artifacts"
+  git push
+
+You can then use `your-experimental-branch-name` as the rev in download
+URLs above.
```
Comment (ichub) [inline README.md:60]: for ppl who don't have write access to this repo, the url will need to refer to a different repo (as PRs for external contributors will be coming from forks, not from this repo). not a huge issue as this readme is probably only intended for internal use anyways.

File: README.md [hunk 6]
```diff
@@ -1,10 +1,78 @@
...
+
+Test artifacts should not be merged into `main`.  Instead the experimental
+branch can be deleted when testing is complete,
+
+## Releasing updated artifacts
+
+To release a new package to NPM you need write access to @pcd packages in NPM.
```
Comment (ichub) [inline README.md]:
```suggestion
To release a new package to NPM you need write access to [`@pcd`](https://www.npmjs.com/search?q=@pcd) packages in NPM.
```

File: README.md [hunk 7]
```diff
@@ -1,10 +1,78 @@
...
+You'll need to run `npm login` once to establish your publishing credentials.
+
+A release can be created using [changesets](https://github.com/changesets/changesets/tree/main).  First declare and describe your intended change:
+
+  yarn changeset
```
Comment (ichub) [inline README.md:74]:
````suggestion
```sh
yarn changeset
```
````

File: README.md [hunk 8]
```diff
@@ -1,10 +1,78 @@
...
+Answer the prompts to select the type of release and description.  Note that
+due to binary compatibility, once artifacts are in production, patches can
+only be used to add new artifacts, or change non-artifact files.  Consider
+using changeset's prerelease or snapshot features for testing first.
+
+Next update the package version and create a commit for the release:
+
+  yarn changeset version
+  git add .
+  git commit
```
Comment (ichub) [inline README.md:87]:
````suggestion
```sh
yarn changeset version
git add .
git commit
```
````

File: README.md [hunk 9]
```diff
@@ -1,10 +1,78 @@
...
+
+Finally publish the release to NPM, which also creates a tag in git, which
+you should push to GitHub:
+
+  yarn changeset publish
+  git push --follow-tags
```
Comment (ichub) [inline README.md:95]:
````suggestion
```sh
yarn changeset publish
git push --follow-tags
```
````

File: package.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "pcd-snark-artifacts-root",
+  "version": "0.0.0",
+  "repository": "https://github.com/proofcarryingdata/snark-artifacts.git",
+  "license": "MIT",
+  "private": true,
```
Comment (ichub) [inline package.json:6]: not sure if this causes problems but suggesting because it makes sense to me

File: packages/proto-pod-gpc/README.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        <img alt="NPM version" src="https://img.shields.io/npm/v/@pcd/proto-pod-gpc-artifacts?style=flat-square" />
+    </a>
+    <a href="https://npmjs.org/package/@pcd/proto-pod-gpc-artifacts">
+        <img alt="Downloads" src="https://img.shields.io/npm/dm/@pcd/proto-pod-gpc-artifacts.svg?style=flat-square" />
+    </a>
+</p>
+
+# `@pcd/proto-pod-gpc-artifacts`
+
+ZK artifacts for supporting the prot-pod-gpc family of general purpose circuits.
```
Comment (ichub) [inline packages/proto-pod-gpc/README.md]:
```suggestion
ZK artifacts for supporting the `proto-pod-gpc` family of general purpose circuits.
```

File: packages/proto-pod-gpc/README.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+These files are needed by code in the [Zupass](https://github.com/proofcarryingdata/zupass)
+repo but are stored separately to avoid bloating the repo size.
+
+These artifacts are used for GPC proofs as generated by the
+[@pcd/gpc](https://github.com/proofcarryingdata/zupass/tree/main/packages/lib/gpc)
```
Comment (ichub) [inline packages/proto-pod-gpc/README.md]:
```suggestion
[`@pcd/gpc`](https://github.com/proofcarryingdata/zupass/tree/main/packages/lib/gpc)
```

File: packages/proto-pod-gpc/README.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+and [@pcd/gpc-pcd](https://github.com/proofcarryingdata/zupass/tree/main/packages/pcd/gpc-pcd)
```
Comment (ichub) [inline packages/proto-pod-gpc/README.md]:
```suggestion
and [`@pcd/gpc-pcd`](https://github.com/proofcarryingdata/zupass/tree/main/packages/pcd/gpc-pcd)
```

File: packages/proto-pod-gpc/README.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+packages.
+
+These artifacts are built from the circom code in the
+[@pcd/gpcircuits](https://github.com/proofcarryingdata/zupass/tree/main/packages/lib/gpcircuits) package.
```
Comment (ichub) [inline packages/proto-pod-gpc/README.md]:
```suggestion
[`@pcd/gpcircuits`](https://github.com/proofcarryingdata/zupass/tree/main/packages/lib/gpcircuits) package.
```

File: packages/proto-pod-gpc/README.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+For info on artifact downloading and publishing, see the documentation in the
+[snark-artifacts](https://github.com/proofcarryingdata/snark-artifacts) repo
```
Comment (ichub) [inline packages/proto-pod-gpc/README.md]:
```suggestion
[`snark-artifacts`](https://github.com/proofcarryingdata/snark-artifacts) repo
```

File: packages/proto-pod-gpc/README.md [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+    </h1>
+</p>
+
+<p align="center">
+    <a href="https://github.com/proofcarryingdata">
+        <img src="https://img.shields.io/badge/project-PCD-blue.svg?style=flat-square">
+    </a>
+    <a href="https://github.com/proofcarryingdata/snark-artifacts/blob/artwyman/experimental/packages/proto-pod-gpc/LICENSE">
+        <img alt="License" src="https://img.shields.io/badge/license-GPL--3.0-green.svg?style=flat-square">
+    </a>
```
Comment (ichub) [inline packages/proto-pod-gpc/README.md]:
```suggestion
    <a href="https://github.com/proofcarryingdata/snark-artifacts/blob/main/packages/proto-pod-gpc/LICENSE">
        <img alt="License" src="https://img.shields.io/badge/license-GPL--3.0-green.svg?style=flat-square">
    </a>
```

**Review by ichub — APPROVED** [review]
nice!

File: package.json (hunk 10 above)
Comment (artwyman) [reply package.json:6]: I think what this does is make sure NPM will refuse to publish this "package", which isn't something we want for the root of the repo.  The package in `packages/proto-pod-pcd` is marked public.

**Pull request merged**